### PR TITLE
Add shaderc bindings

### LINKED ISSUE #482: Add Shaderc and  SPIRV-Cross bindings?
Hello,  I would like to know if there are any plans to add [Shaderc ](https://github.com/google/shaderc) and [SPIRV-Cross](https://github.com/KhronosGroup/SPIRV-Cross).

### PR DESCRIPTION
**Pull request by Beyley**
# Summary of the PR
Adds Shaderc bindings

# Related issues, Discord discussions, or proposals
Closes #482

### CODE REVIEW

**Review by Perksey — COMMENTED** [review]
Bindings portion is approved

**Review by hYdos — APPROVED** [review]
this looks good. whats it waiting on?

**Comment by Beyley** [comment]
> this looks good. whats it waiting on?

Native libraries, i have the Zig build script to do this, i just need to hook it up to our CI, and i have not found the time/energy to do so yet
[build.zig.txt](https://github.com/dotnet/Silk.NET/files/11780199/build.zig.txt)

File: build/nuke/Native/SDL2.cs [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+                    InheritedShell($"{envVars64bit} ./configure --prefix={x64BuildDir}", SDL2Path).AssertZeroExitCode();
+                    InheritedShell($"{envVars64bit} make {JobsArg}", SDL2Path).AssertZeroExitCode();
+                    InheritedShell($"make install", SDL2Path).AssertZeroExitCode();
+
+                    //Strip the libraries
+                    InheritedShell($"strip {x86BuildDir / "lib" / "libSDL2-2.0.so.0.2600.5"}", SDL2Path).AssertZeroExitCode();
+                    InheritedShell($"strip {x64BuildDir / "lib" / "libSDL2-2.0.so.0.2600.5"}", SDL2Path).AssertZeroExitCode();
+
+                    CopyFile(x86BuildDir / "lib" / "libSDL2-2.0.so.0.2600.5", runtimes / "linux-x86" / "native" / "libSDL2-2.0.so", FileExistsPolicy.Overwrite);
```
[the remark this answers is not in the document]
Comment (Beyley) [reply build/nuke/Native/SDL2.cs]: it has been made Globby

File: build/nuke/Native/SDL2.cs [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+            }
+
+            if(OperatingSystem.IsLinux())
+            {
+                if(RuntimeInformation.OSArchitecture == Architecture.Arm64) 
+                {
+                    InheritedShell("cmake ..", x86BuildDir).AssertZeroExitCode();
+                    InheritedShell("cmake --build .", x86BuildDir).AssertZeroExitCode();
+
+                    CopyFile(ARM64BuildDir / "libSDL2-2.0.so.0.2600.5", runtimes / "linux-arm64" / "native" / "libSDL2-2.0.so", FileExistsPolicy.Overwrite);
```
Comment (Perksey) [inline build/nuke/Native/SDL2.cs:74]: Tlobby here too

**Review by Perksey — CHANGES_REQUESTED** [review]

**Review by Perksey — APPROVED** [review]
Dtunbjeb approval

**Comment by Beyley** [comment]
Due to broken CLA check, merging manually

**Pull request merged**